torch/onnx/utils.py

```diff
@@ -172,6 +172,20 @@ def setup_onnx_logging(verbose: bool):
             torch.onnx.disable_log()
 
 
+@contextlib.contextmanager
+@_beartype.beartype
+def _setup_onnx_diagnostic():
```

Comment:
```suggestion
def _get_onnx_diagnostic():
```
`_setup_onnx_diagnostic() as diagnostic_ctx` feels a bit odd as `setup` routines usually do not return objects

Reply:
+1

Reply:
`_get_diagnostic_context` etc.

Reply:
how about `_create_diagnostic_context`?

Reply:
I like it